src/main/java/sirius/biz/jobs/FixProcessTypes.java

```diff
@@ -0,0 +1,134 @@
+import sirius.db.es.Elastic;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.PriorityParts;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.nls.NLS;
+import sirius.web.security.Permission;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+/**
+ * Provides a job which fixes process types to now used getName() instead of the NLS-key as processType.
+ */
+@Register(framework = Processes.FRAMEWORK_PROCESSES)
+@Permission(TenantUserManager.PERMISSION_SYSTEM_ADMINISTRATOR)
+public class FixProcessTypes extends SimpleBatchProcessJobFactory {
+
+    @PriorityParts(JobFactory.class)
+    private List<JobFactory> factories;
+
+    @Part
+    private Elastic elastic;
+
+    private static final Parameter<Boolean> SIMULATION =
+            new BooleanParameter("simulation", "Simulation").withDescription(
+                                                                    "Simulation mode will only output which keys are missing and how many processes would have their type changed.")
+                                                            .withDefaultTrue()
+                                                            .build();
+
+    @Override
+    protected void collectParameters(Consumer<Parameter<?>> parameterCollector) {
+        parameterCollector.accept(SIMULATION);
+    }
+
+    @Override
+    protected void execute(ProcessContext process) throws Exception {
+        boolean simulation = process.require(SIMULATION);
+
+        Map<String, String> jobTypesAndKeys = factories.stream()
+                                                       .filter(JobFactory::canStartInBackground)
+                                                       .distinct()
+                                                       .collect(HashMap::new,
+                                                                (map, job) -> map.put(job.getName(),
+                                                                                      job.getClass().getSimpleName()
+                                                                                      + ".label"),
+                                                                HashMap::putAll);
+
+        process.log(ProcessLog.info().withMessage(Strings.apply("Found %s jobs", jobTypesAndKeys.size())));
+        jobTypesAndKeys.forEach((type, nlsKey) -> {
+            AtomicInteger counter = new AtomicInteger();
+            if (!NLS.exists(nlsKey, null)) {
+                process.log(ProcessLog.warn()
+                                      .withMessage(Strings.apply("No NLS key found for type %s (%s)", type, nlsKey)));
+            }
+
+            elastic.select(Process.class).eq(Process.PROCESS_TYPE, nlsKey).iterateAll(executedProcess -> {
```

Comment:
I think its smarter to streamBlockwise all processes. Per process you lookup the right key for it (and add to a local cache). 
Streamblockwise sorts per ID and is abortable. If the job stops, print the last process ID and add an optional param to resume from the given optional ID (ID >= x). In OX we are talking about 580K procs!
And use the progress tracker to have a hint of execution time.

Reply:
Also consider already migrated jobs and skip the ones with processType.endsWith(".label"). This will also discard new jobs started after the change.

Reply:
All Jobs which have been migrated already or executed after the change will not be checked a second time, because the processType is no longer MyJobFactory.label
A crash might be possible, maybe a count in the check-mode will be safer than iterating.